src/Workspaces/CSharp/Portable/CodeGeneration/ParameterGenerator.cs

```diff
@@ -140,14 +141,34 @@ internal static ParameterSyntax GetParameter(IParameterSymbol p, CodeGenerationO
         }
 
         private static SyntaxList<AttributeListSyntax> GenerateAttributes(
-            IParameterSymbol parameter, bool isExplicit, CodeGenerationOptions options)
+            IParameterSymbol parameter, bool isExplicit, CodeGenerationOptions options, ref NullableAnnotation nullableAnnotation)
         {
             if (isExplicit)
             {
                 return default;
             }
 
             var attributes = parameter.GetAttributes();
+
+            if (parameter.Type.IsReferenceType)
+            {
+                var newAttributes = attributes.RemoveAll(IsAllowNullAttribute);
```

Comment:
What about DisallowNull?

Reply:
[DisallowNull] and [NotNull] are preserved for reference types. See #39256.

Reply:
Added.